presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -123,6 +127,12 @@ public void registerQuery(SqlQueryExecution sqlQueryExecution, SubPlan fragmente
                     dynamicFilters,
                     lazyDynamicFilters,
                     replicatedDynamicFilters);
+
+            sqlQueryExecution.addStateChangeListener(state -> {
+                if (state == STARTING) {
+                    queryIsStarting(sqlQueryExecution.getQueryId());
```

Comment:
The query will be STARTING very soon:
https://github.com/prestosql/presto/blob/a29fc64a478818c52c0714e05d24241cc6ef4464/presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java#L376-L379

the callback adds complexity & theoretical uncertainty whcih i would prefer to avoid, if possible.
would it be still OK to start the timer immediately here?